classes/battle/BattleManager.php

```diff
@@ -651,6 +651,66 @@ private function checkForWinner(): string {
         return $this->battle->winner;
     }
 
+    #[Trace]
+    private function handleRoundCompletion(string $round_winner): string {
+        // if round winner is team1, increment wins
+        if ($round_winner == Battle::TEAM1) {
+            $this->battle->team1_wins++;
+        }
+        // if round winner is team2, increment wins
+        if ($round_winner == Battle::TEAM2) {
+            $this->battle->team2_wins++;
+        }
+        // if single-round battle and round is draw
+        if ($this->battle->rounds <= 1 && $round_winner == Battle::DRAW) {
+            return Battle::DRAW;
+        }
+        // if team1 majority of wins (total rounds)
+        if ($this->battle->team1_wins > floor($this->battle->rounds / 2) || ($this->battle->round_count >= $this->battle->rounds && $this->battle->team1_wins > $this->battle->team2_wins)) {
+            return Battle::TEAM1;
+        }
+        // if team2 majority of wins (total rounds)
+        if ($this->battle->team2_wins > floor($this->battle->rounds / 2) || ($this->battle->round_count >= $this->battle->rounds && $this->battle->team2_wins > $this->battle->team1_wins)) {
+            return Battle::TEAM2;
+        }
+        // if more rounds to go
+        if ($this->battle->round_count < $this->battle->rounds) {
+            $this->battle->round_count++;
+            $this->battle->player1->health = $this->battle->player1->max_health;
+            $this->battle->fighter_health[$this->battle->player1->combat_id] = $this->battle->player1->health;
+            $this->battle->player1_last_damage_taken = 0;
+            $this->battle->player2->health = $this->battle->player2->max_health;
+            $this->battle->fighter_health[$this->battle->player2->combat_id] = $this->battle->player2->health;
+            $this->battle->player2_last_damage_taken = 0;
+            $this->effects->active_effects = [];
+            $this->battle->jutsu_cooldowns = [];
+            $this->battle->turn_count = 0;
+            $this->battle->turn_time = time();
+            $this->battle->player1_time = Battle::MAX_TURN_LENGTH;
+            $this->battle->player2_time = Battle::MAX_TURN_LENGTH;
```

Comment:
This looks like the same logic as below, could you extract it to a function?

Reply:
Yeah both cases there are completely identical (for now) - I'll copy this into a helper function (`resetBattle()`) and we can define differences with how extra rounds work separately.